skoolkit/basic.py

```diff
@@ -114,14 +115,92 @@
     255: 'COPY'
 }
 
+def get_number(snapshot, i):
+    if snapshot[i] == 0:
+        # Small integer (signed)
+        num = _get_integer(snapshot, i)
+    else:
+        # Floating point number (signed)
+        num = _get_float(snapshot, i)
+    return num
+
+def _get_integer(snapshot, i):
+    # See ZX Spectrum ROM routine INT_FETCH at 0x2D7F
+    sign = snapshot[i + 1]
+    lsb = (snapshot[i + 2] ^ sign) - sign
+    if lsb < 0:
+        lsb += 256
+        carry = 1
+    else:
+        carry = 0
+
+    msb = ((snapshot[i + 3] + sign + carry) & 255) ^ sign
+
+    num = 256 * msb + lsb
+    if (sign & 128) != 0:
+        num *= -1
+
+    return num
+
+def _get_float(snapshot, i):
+    exponent = snapshot[i] - 160
+    if (snapshot[i + 1] & 128) == 0:
+        sign = 1
+    else:
+        sign = -1
```

Comment:
Pythonicity applied and redundant brackets removed in commit 0e626d6530021a525722c4f139968c439404b5a9.